subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultWindowsSdkLocator.java

```diff
@@ -13,270 +13,55 @@
-            }
-        }
-        return candidate == null ? new SdkNotFound("Could not locate a Windows SDK installation, using the Windows registry and system path.") : new SdkFound(candidate);
-    }
-
-    private void addSdk(File path, String version, String name) {
-        foundSdks.put(path, new WindowsSdk(path, VersionNumber.parse(version), name));
-    }
-
-    private static boolean isWindowsSdk(File candidate) {
-        boolean hasResourceCompiler = false;
-        boolean hasKernel32Lib = false;
-
-        for (String path : RESOURCE_PATHS) {
-            if (new File(candidate, path + RESOURCE_FILENAME).isFile()) {
-                hasResourceCompiler = true;
-                break;
-            }
-        }
-
-        for (String path : KERNEL32_PATHS) {
-            if (new File(candidate, path + KERNEL32_FILENAME).isFile()) {
-                hasKernel32Lib = true;
-                break;
-            }
-        }
-
-        return hasResourceCompiler && hasKernel32Lib;
-    }
-
-    private static String formatVersion(String version) {
-        int index = StringUtils.ordinalIndexOf(version, ".", 2);
-
-        if (index != -1) {
-            version = version.substring(0, index);
-        }
-
-        return version;
-    }
-
-    private static class SdkFound implements SearchResult {
-        private final WindowsSdk sdk;
-
-        public SdkFound(WindowsSdk sdk) {
-            this.sdk = sdk;
+        SdkSearchResult(SearchResult legacySearchResult, WindowsKitComponentLocator.SearchResult<WindowsKitWindowsSdk> windowsKitSearchResult) {
+            this.legacySearchResult = legacySearchResult;
+            this.windowsKitSearchResult = windowsKitSearchResult;
         }
 
         @Override
         public WindowsSdk getSdk() {
-            return sdk;
-        }
-
-        @Override
-        public boolean isAvailable() {
-            return true;
-        }
-
-        @Override
-        public void explain(TreeVisitor<? super String> visitor) {
-        }
-    }
-
-    private static class SdkNotFound implements SearchResult {
-        private final String message;
-
-        private SdkNotFound(String message) {
-            this.message = message;
-        }
-
-        @Override
-        public WindowsSdk getSdk() {
-            return null;
+            if (windowsKitSearchResult.isAvailable()) {
+                return windowsKitSearchResult.getComponent();
+            } else if (legacySearchResult.isAvailable()) {
+                return legacySearchResult.getSdk();
+            } else {
+                return null;
+            }
         }
 
         @Override
         public boolean isAvailable() {
-            return false;
+            return windowsKitSearchResult.isAvailable() || legacySearchResult.isAvailable();
         }
 
         @Override
         public void explain(TreeVisitor<? super String> visitor) {
-            visitor.node(message);
+            legacySearchResult.explain(visitor);
```

Comment:
Should we conditionally explain Windows Kit search if available else use the legacy one?

Reply:
This only gets called if neither is found (i.e. if `isAvailable()` returns false).  Both of the locators return basically the same message (i.e. "I couldn't find a windows sdk anywhere").  I figured it made sense to just have it return the same message what it always has (i.e. the legacy message).